cmd/frontend/graphqlbackend/campaigns.go

```diff
@@ -92,97 +51,64 @@ type FileDiffsConnectionArgs struct {
-	PatchSetByID(ctx context.Context, id graphql.ID) (PatchSetResolver, error)
-
-	PatchByID(ctx context.Context, id graphql.ID) (PatchInterfaceResolver, error)
+type CreateChangesetSpecArgs struct {
+	ChangesetSpec string
 }
 
-var campaignsOnlyInEnterprise = errors.New("campaigns and changesets are only available in enterprise")
-
-type defaultCampaignsResolver struct{}
+type CreateCampaignSpecArgs struct {
+	Namespace graphql.ID
 
-var DefaultCampaignsResolver CampaignsResolver = defaultCampaignsResolver{}
-
-func (defaultCampaignsResolver) CreateCampaign(ctx context.Context, args *CreateCampaignArgs) (CampaignResolver, error) {
-	return nil, campaignsOnlyInEnterprise
+	CampaignSpec   string
+	ChangesetSpecs []graphql.ID
 }
 
-func (defaultCampaignsResolver) UpdateCampaign(ctx context.Context, args *UpdateCampaignArgs) (CampaignResolver, error) {
-	return nil, campaignsOnlyInEnterprise
-}
-
-func (defaultCampaignsResolver) CampaignByID(ctx context.Context, id graphql.ID) (CampaignResolver, error) {
-	return nil, campaignsOnlyInEnterprise
-}
-
-func (defaultCampaignsResolver) Campaigns(ctx context.Context, args *ListCampaignArgs) (CampaignsConnectionResolver, error) {
-	return nil, campaignsOnlyInEnterprise
-}
-
-func (defaultCampaignsResolver) DeleteCampaign(ctx context.Context, args *DeleteCampaignArgs) (*EmptyResponse, error) {
-	return nil, campaignsOnlyInEnterprise
-}
+type CampaignsResolver interface {
+	// Mutations
+	CreateCampaign(ctx context.Context, args *CreateCampaignArgs) (CampaignResolver, error)
+	ApplyCampaign(ctx context.Context, args *ApplyCampaignArgs) (CampaignResolver, error)
+	MoveCampaign(ctx context.Context, args *MoveCampaignArgs) (CampaignResolver, error)
+	CloseCampaign(ctx context.Context, args *CloseCampaignArgs) (CampaignResolver, error)
+	DeleteCampaign(ctx context.Context, args *DeleteCampaignArgs) (*EmptyResponse, error)
+	CreateChangesetSpec(ctx context.Context, args *CreateChangesetSpecArgs) (ChangesetSpecResolver, error)
+	CreateCampaignSpec(ctx context.Context, args *CreateCampaignSpecArgs) (CampaignSpecResolver, error)
+	// ComputeCampaignDelta(ctx context.Context, args *ComputeCampaignDeltaArgs) (CampaignDeltaResolver, error)
+	SyncChangeset(ctx context.Context, args *SyncChangesetArgs) (*EmptyResponse, error)
 
-func (defaultCampaignsResolver) RetryCampaignChangesets(ctx context.Context, args *RetryCampaignChangesetsArgs) (CampaignResolver, error) {
-	return nil, campaignsOnlyInEnterprise
+	// Queries
+	Campaigns(ctx context.Context, args *ListCampaignArgs) (CampaignsConnectionResolver, error)
+	CampaignByID(ctx context.Context, id graphql.ID) (CampaignResolver, error)
+	ChangesetByID(ctx context.Context, id graphql.ID) (ChangesetResolver, error)
 }
 
-func (defaultCampaignsResolver) CloseCampaign(ctx context.Context, args *CloseCampaignArgs) (CampaignResolver, error) {
-	return nil, campaignsOnlyInEnterprise
-}
+type CampaignSpecResolver interface {
+	ID() (graphql.ID, error)
```

Comment:
I think this should be, otherwise the Node interface is not fulfilled


```suggestion
	ID() graphql.ID
```